lib/fibonacci.rb

```diff
@@ -1,8 +1,21 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n) where n is n
+# Space Complexity - O(1) where the size of memo_arr is always 2
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This is technically still O(n) space complexity due to the call stack.  Otherwise yes.